provisioning/transport/amqp/devdoc/amqp_requirements.md

```diff
@@ -55,6 +55,10 @@ iotdps-forceRegistration: <true or false>;
 ```
  **]**
 
+**SRS_NODE_PROVISIONING_AMQP_06_007: [** If the `registrationRequest` send request is rejected with an `InternalError` or `ThrottlingError`, the result.status value will be set with `registering` and the callback will be invoked with *no* error object. **]**
+
```

Comment:
we've already talked about this and i still don't feel comfortable about the fact that we're pushing down what is effectively a retry policy in the transport.
retry policies should have a max retry count or max retry time. should be configurable. should be consistent across our clients.

this is effectively mounting technical debt that you're going to have to manage.

Reply:
In fact the exact opposite is occurring.  The transports make absolutely no retry decisions at all.  They simply acknowledge the fact that dps service has extended/defined more precisely what a fatal error is.  All the transports are doing are intpreting the protocol specific ways it is expressed by the service and turned them into agnostic presentation to the polling machine.  I'll go further, even the polling machine isn't defining what the retry strategy is.  That was and still remains with the dps service.  The dps service has the ability to return a FAIL status at ANY time.  If it does, it will be propagated with zero change in the polling or transport code.  In addition the service will effectively fail this anyway after about twenty minutes.  That is my understanding anyway.  Not like there is a precise contract between clients and service even now.  I've done my best to communicate what the precise definitions are to the rest of the SDK team.

I actually also disagree with the whole retry counts up in the higher provisioning client layer.  Again, the service is nicely in control here.  Like we try not validate the inner semantics of a message to the services and let them evaluate them, since the service is offering to make the decision on how to proceed with the request we absolutely should take it.  THAT provides consistency across SDKs.

---
In reply to: [282199689](https://github.com/Azure/azure-iot-sdk-node/pull/555#discussion_r282199689) [](ancestors = 282199689)